lib/foreman/logging.rb

```diff
@@ -0,0 +1,159 @@
+require 'logging'
+require 'fileutils'
+
+module Foreman
+  class LoggingImpl
+    private_class_method :new
+
+    attr_reader :config, :log_directory
+
+    def configure(options = {})
+      fail 'logging can be configured only once' if @configured
+      @configured = true
+
+      @log_directory = options.fetch(:log_directory, './log')
+      ensure_log_directory(@log_directory)
+
+      load_config(options.fetch(:environment))
+
+      configure_color_scheme
+      configure_root_logger(options)
+
+      build_console_appender
+    end
+
+    def add_loggers(loggers = {})
+      return unless loggers.is_a?(Hash)
+
+      loggers.each do |name, config|
+        add_logger(name, config)
+      end
+    end
+
+    def add_logger(logger_name, logger_config)
+      logger          = ::Logging.logger[logger_name]
+      logger.level    = logger_config[:level] if logger_config.key?(:level)
+      logger.additive = logger_config[:enabled] if logger_config.key?(:enabled)
+      logger.trace    = @config[:log_trace]
```

Comment:
Answering my own question about 2.0.0, this causes a small problem here and on the root logger: https://github.com/TwP/logging/commit/5c06041

If we could support both, that'd be appreciated, until we can get rid of our older logging RPM.